src/Symfony/Component/Validator/Mapping/PropertyMetadata.php

```diff
@@ -50,8 +50,21 @@ public function getPropertyValue($object)
     {
         $reflProperty = $this->getReflectionMember($object);
 
-        if (\PHP_VERSION_ID >= 70400 && !$reflProperty->isInitialized($object)) {
-            return null;
+        if (\PHP_VERSION_ID >= 70400 && $reflProperty->hasType() && !$reflProperty->isInitialized($object)) {
+            // There is no way to check if a property has been unset or it is uninitialized.
+            // When trying to access an uninitialized property, __get method is triggered in PHP 7.4.0, but not in 7.4.1+.
```

Comment:
we shouldn't mention any versions of PHP in the comment
we should make the code work consistently in all versions as much as possible.